# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/painting/nsDisplayList.cpp
# Commit: 5ded58486aba
# Full Hash: 5ded58486abab5712f305a88bbdd746ae74dcdef
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2023-03-06 09:45:20
# Description:
#   Bug 1819957 - Split out nsTextPaintStyle into its own source file. r=emilio
#   
#   This lets us instantiate it on the stack in nsDisplayText::Paint.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171635
# ==============================================================================

diff -r 58235f400c11 -r 5ded58486aba layout/painting/nsDisplayList.cpp
--- a/layout/painting/nsDisplayList.cpp	Sun Mar 05 20:37:07 2023 +0000
+++ b/layout/painting/nsDisplayList.cpp	Sun Mar 05 22:24:10 2023 +0000
@@ -97,6 +97,7 @@
 #include "nsTableCellFrame.h"
 #include "nsTableColFrame.h"
 #include "nsTextFrame.h"
+#include "nsTextPaintStyle.h"
 #include "nsSliderFrame.h"
 #include "nsFocusManager.h"
 #include "TextDrawTarget.h"
@@ -7558,8 +7559,9 @@
 
   // Similarly for shadows that may be cast by ::selection.
   if (f->IsSelected()) {
+    nsTextPaintStyle textPaint(f);
     Span<const StyleSimpleShadow> shadows;
-    f->GetSelectionTextShadow(SelectionType::eNormal, &shadows);
+    f->GetSelectionTextShadow(SelectionType::eNormal, textPaint, &shadows);
     addShadowSourceToVisible(shadows);
   }
 
